atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/ConflictHandler.java

```diff
@@ -60,13 +60,76 @@
      * This should not be used on tables that are expected to have tons of reads because we must keep the whole read
      * set in memory so we can compare it at commit time to ensure that everything we read has not changed.
      */
-    SERIALIZABLE,
+    SERIALIZABLE(false, true, true, true),
 
     /**
      * Same as {@link RETRY_ON_WRITE_WRITE}, but checks for conflicts by locking cells during commit instead
      * of locking rows. Cell locks are more fine-grained, so this will produce less contention at the expense
      * of requiring more locks to be acquired.
      */
-    RETRY_ON_WRITE_WRITE_CELL;
+    RETRY_ON_WRITE_WRITE_CELL(true, false, true, false),
 
+    /**
+     * Same as {@link SERIALIZABLE}, but checks for conflicts by locking cells during commit instead
+     * of locking rows. Cell locks are more fine-grained, so this will produce less contention at the expense
+     * of requiring more locks to be acquired.
+     */
+    SERIALIZABLE_CELL(true, false, true, true),
+
+    /**
+     * This conflict handler is designed to be used by index tables. As any write/write conflict on an index table
+     * will necessarily also be a write/write conflict on base table, we do not need to check write/write conflicts
+     * on index tables. Read/write conflicts should still need to be checked, since we do not need to read the index
+     * table with the main table.
+     * <p>
+     * This conflict handler locks at cell level. As index tables use dynamic columns, locking at row level would
+     * result in higher contention.
+     */
+    SERIALIZABLE_INDEX(true, false, false, true),
+
+    /**
+     * This conflict handler is designed to be used for migrating a table from SERIALIZABLE to SERIALIZABLE_CELL
+     * conflict handler, or vice versa without requiring a shutdown.
```

Comment:
simultaneous shutdown of all clients

Reply:
Maybe also clarify that once all nodes are on `SERIALIZABLE_LOCK_LEVEL_MIGRATION`, then can move to the desired end state